text/0000-opentelemetry-protocol.md

```diff
@@ -0,0 +1,393 @@
+
+If the client shuts down before receiving acknowledging responses from the server or otherwise has to discard the requests that are in `AckPending` it may assume that telemetry data contained in these requests is lost. Client will typically maintain a counter of lost telemetry data.
+
+### Multi-Destination Exporting
+
+When the telemetry data from one client must be sent to more than one destination server there is an additional complication that must be accounted for. When one of the servers acknowledges the data and the other server does not (yet) acknowledges the client needs to make a decision about how to move forward.
+
+In such situation the the client SHOULD implement queuing, acknowledgement handling and retrying logic per destination. This ensures that servers do not block each other. The queues SHOULD reference shared, immutable data to be sent, thus minimizing the memory overhead caused by having multiple queues.
+
+![Multi-Destination Exporting](images/otlp-multi-destination.png)
+
+This ensures that all destination servers receive the data regardless of their speed of reception (within the available limits imposed by the size of the client-side queue).
+
+## Trade-offs and mitigations
+
+### Acknowledgements
+
+#### Implementation Complexity
+
+The addition of acknowledgements provides delivery guarantees however it comes at the cost of slightly increased processing overhead and complexity of implementation.
+
+#### Duplicate Data
+
+In edge cases (e.g. on reconnections, network interruptions, etc) the client has no way of knowing if recently sent data was delivered if no acknowledgement was received yet. The client will typically choose to re-send such data to guarantee delivery, which may result in duplicate data on the server side. This is a deliberate choice and is considered to be the right tradeoff for telemetry data.
+
+### Client Mode
+
+Clients that want to avoid complex implementations can choose to implement Synchronous mode only.
+
+## Future Versions and Compatibility
+
+OTLP will evolve and change over time. Future versions of OTLP must be designed and implemented in a way that ensures that clients and servers that implement different versions of OTLP can interoperate and exchange telemetry data. Old clients must be able to talk to new servers and vice versa. If new versions of OTLP introduce new functionality that cannot be understood and supported by nodes implementing the old versions of OTLP the protocol must regress to the lowest common denominator from functional perspective.
+
+Future versions MUST ensure this interoperability between nodes with versions that are apart at least one major version number. When possible such interoperability SHOULD be ensured between all versions of OTLP that are not declared obsolete.
+
+OTLP does not use explicit version numbering in the messages or in gRPC service calls. OTLP's interoperability of different versions is based on 2 concepts:
```

Comment:
Is this the right approach? What's wrong with using a HELLO request/response where client and server explicitly exchange their protocol version numbers?